pkg/compactor/bucket_compactor.go

```diff
@@ -857,7 +860,10 @@ func (c *BucketCompactor) Compact(ctx context.Context, maxCompactionTime time.Du
 							"UnhealthyBlock: marking unhealthy block as no compact to unblock compaction",
 							c.metrics.blocksMarkedForNoCompact.WithLabelValues(block.CriticalNoCompactReason),
 						)
-						if err == nil {
+						if err != nil && errors.As(err, &block.TSDBBlockRecoverableError{}) {
```

Comment:
it is needed because of this `As will panic if target is not a non-nil pointer to either a type that implements error, or to any interface type. As returns false if err is nil.` the unittest failed, that is why i added the error check